homeassistant/components/garadget/cover.py

```diff
@@ -213,23 +213,20 @@ def _check_state(self, now):
     def close_cover(self, **kwargs: Any) -> None:
         """Close the cover."""
         if self._state not in ["close", "closing"]:
-            ret = self._put_command("setState", "close")
+            self._put_command("setState", "close")
             self._start_watcher("close")
-            return ret.get("return_value") == 1
```

Comment:
I'm surprised that this passed `mypy`...
It previously returned a `bool` even though the return type is set to `None`
Maybe @cdce8p has an explaination?

Reply:
`ret` is the return value of the unannotated `self._put_command` method, i.e. inferred as `Any`. Furthermore, any operation with `Any` will also result in `Any` and by definition `Any` is compatible with `None`.

```py
reveal_type(ret)
reveal_type(ret.get("return_value"))
reveal_type(ret.get("return_value") == 1)
```
```
homeassistant/components/garadget/cover.py:218: note: Revealed type is "Any"
homeassistant/components/garadget/cover.py:219: note: Revealed type is "Any"
homeassistant/components/garadget/cover.py:220: note: Revealed type is "Any"
```

Reply:
Thanks, makes sense!
I'm surprised that `Any == 1` reveals as `Any` - I would have expected this to reveal as `bool` but hey...